### PR TITLE
autoload color palette. closes #138

### LINKED ISSUE #138: Airline Theme Contains Error
### What happened

Using `:AirlineTheme dracula` returns

```
airline: There is an error in theme "dracula".
```

### What I expected to happen

Airline theme is applied.

### Screenshot

image

### Machine Info

- **Vim type (`vim`/`gvim`/`neovim`)**: gvim
- **Vim version**: 8.1.1 (installed via Chocolatey)
- **OS**: Win 10 Pro x64 Version 1809 (OS Build 17763.529)
- **Terminal/Terminal Emulator/VTE**: n/a
- **`TERM` environment variable**: n/a

### Additional Info

I have the dracula-theme, vim-airline, and vim-airline-themes plugins installed via pathogen. The Airline theme for dracula is being loaded from its autoload/airline/themes directory. If I rename that file the dracula theme delivered with vim-airline-themes gets loaded instead without errors.

I ran `:debug :AirlineTheme dracula` to determine this. Stepping through the code reveals this:

image

What am I missing?

### PR DESCRIPTION
**Pull request by dsifford**
See #138 for more details on this.

Aside from the obvious, I also trimmed down a couple pieces of misc cruft here and there, and also adjusted some whitespace issues.

Let me know if you have any questions.

Ping @benknoble

### CODE REVIEW

**Comment by dsifford** [comment]
Haven't heard back. Gonna merge this as it works fine for me everywhere I've tried it.

File: autoload/airline/themes/dracula.vim [hunk 1]
```diff
@@ -14,14 +14,10 @@
...
-  if ! type(l:fg) == 3
-    echoerr 'Color undefined in dracula_palette: ' . a:fg
-  elseif ! type(l:bg) == 3
-    echoerr 'Color undefined in dracula_palette: ' . a:fg
-  endif
-  return [ l:fg[0], l:bg[0], l:fg[1], l:bg[1] ] + filter(copy(a:000), 'type(v:val) == 1 && len(v:val) > 0')
+  let l:fg = g:dracula#palette[a:fg]
+  let l:bg = g:dracula#palette[a:bg]
+  return [ l:fg[0], l:bg[0], l:fg[1], l:bg[1] ] +
+       \ filter(copy(a:000), {_, val -> type(val) == 1 && len(val) > 0 })
```
Comment (benknoble) [inline autoload/airline/themes/dracula.vim]: Been on vacation for a bit, so just seeing this now. Everything looks good, although I can see this lambda syntax being an issue for older vims (which don't support it).

I don't *mind* not supporting older vims, but we should be explicit about it (and announce some kind of breaking change/increment a version number/whatever we have to do). We may want to solicit feedback about it.

Supporting older vims is as simple as converting the lambda back to a string using `v:val`.

Comment (dsifford) [reply autoload/airline/themes/dracula.vim]: We can revert that line. Not worth breaking over something so trivial.

**Comment by jsit** [comment]
Where did these extra ANSI colors come from, I'm curious? I don't see them documented anywhere.

**Comment by dsifford** [comment]
@jsit [here](https://dsifford.github.io/dracula-spec/#sec-ANSI) per ongoing discussion [here](https://github.com/dracula/dracula-theme/issues/232)

**Pull request merged**